nmi_mysql/nmi_mysql.py

```diff
@@ -9,9 +9,42 @@
 from queue import Queue
 
 
+CONFIG_KEYS = ['host', 'user', 'password', 'db', 'port']
+MAX_POOL_SIZE = 10
+
+
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf):
+        for c in CONFIG_KEYS:
+            if c not in conf:
+                raise ValueError('Invalid config object')
+
+        self.conf = conf
+        self.max_pool_size = conf.get('max_pool_size', MAX_POOL_SIZE)
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put(_Connection(self.conf))
+
+    def query(self, query, params=None):
+        con = self.pool.get()
+        con.connect()
```

Comment:
I talked to ninz and the reason why we built this is because pymysql lacks a good documentation. Also, we wanted to pattern it to anytv-node-mysql to achieve a more elegant code that will be easily used by the developers.
Also, for the libraries you suggested,
- pysqlpool only uses python2
- sqlalchemy is not simple since it's a big library (includes ORM) and you need more lines of code which we're avoiding since we want to have a clean and elegant code.

I'm thinking of changing this project's base to sqlalchemy instead of pymysql but it needs more time to be rewritten rather than continuing our pool implementation with this one.